cnd/src/swap_protocols/rfc003/create_swap.rs

```diff
@@ -138,9 +151,12 @@ where
 /// Returns a future that waits for events on beta ledger to happen.
 ///
 /// Each event is yielded through the controller handle (co) of the coroutine.
+#[allow(clippy::type_complexity)]
 async fn watch_beta_ledger<D, AL, AA, BL, BA>(
```

Comment:
As this is a private fn, you could potentially introduce more type parameters. Not sure if it helps.

Reply:
Similar to #2069 this would need to link associated types of one type param with another type param:
```
172 |     co.yield_(SwapEvent::BetaDeployed(deployed.clone())).await;
    |                                       ^^^^^^^^^^^^^^^^ expected type parameter `BT`, found associated type
```

I'll go for a type alias instead and stop here.